api/src/services/script-runner.service.ts

```diff
@@ -28,34 +31,541 @@ export class ScriptRunnerService {
+              select: {
+                id: true,
+              },
+              where: {
+                name: unit['name'],
+              },
+            });
+          }
+          if (unit['priority_type_id']) {
+            const fullPriorityType = priorityTypes.find(
+              (type) => type.id === unit['priority_type_id'],
+            );
+            priorityType = doorwayPriorityTypes.find(
+              (type) => type.name === fullPriorityType.name,
+            );
+          }
+          if (unit['unit_rent_types']) {
+            const fullRentType = rentTypes.find(
+              (type) => type.id === unit['unit_rent_types'],
+            );
+            rentType = doorwayRentTypes.find(
+              (type) => type.name === fullRentType.name,
+            );
+          }
+
+          await this.prisma.units.create({
+            data: {
+              listings: {
+                connect: {
+                  id: createdListing.id,
+                },
+              },
+              amiChart: doorwayAmiChart
+                ? {
+                    connect: {
+                      id: doorwayAmiChart.id,
+                    },
+                  }
+                : undefined,
+              unitTypes: unitType
+                ? {
+                    connect: {
+                      id: unitType.id,
+                    },
+                  }
+                : undefined,
+              unitAccessibilityPriorityTypes: priorityType
+                ? { connect: { id: priorityType.id } }
+                : undefined,
+              unitRentTypes: rentType
+                ? { connect: { id: rentType.id } }
+                : undefined,
+              amiPercentage: unit['ami_percentage'],
+              annualIncomeMin: unit['annual_income_min'],
+              annualIncomeMax: unit['annual_income_max'],
+              floor: unit['floor'],
+              monthlyIncomeMin: unit['monthly_income_min'],
+              maxOccupancy: unit['max_occupancy'],
+              minOccupancy: unit['min_occupancy'],
+              monthlyRent: unit['monthly_rent'],
+              numBathrooms: unit['num_bathrooms'],
+              numBedrooms: unit['num_bedrooms'],
+              number: unit['number'],
+              sqFeet: unit['sq_feet'],
+              monthlyRentAsPercentOfIncome:
+                unit['monthly_rent_as_percent_of_income'],
+              bmrProgramChart: unit['bmr_program_chart'],
+              status: unit['status'],
+            },
+          });
+        });
+
+        // Migrate the listing to multiselect question mapping
+        const listingMultiselectQuestions: {
+          ordinal: number;
+          listing_id: string;
+          multiselect_question_id: string;
+        }[] = await client.$queryRawUnsafe(
+          `SELECT * FROM listing_multiselect_questions WHERE listing_id = '${listing['id']}'`,
+        );
+        if (listingMultiselectQuestions?.length) {
```

Comment:
This section looks missing from tests too

Reply:
new test added!